pydantic/_internal/_generate_schema.py

```diff
@@ -841,20 +843,36 @@ def _typed_dict_schema(self, typed_dict_cls: Any, origin: Any) -> core_schema.Co
             self.defs.definitions[typed_dict_ref] = schema
             return core_schema.definition_reference_schema(typed_dict_ref)
 
-    def _namedtuple_schema(self, namedtuple_cls: Any) -> core_schema.CallSchema:
+    def _namedtuple_schema(self, namedtuple_cls: Any, origin: Any) -> core_schema.CoreSchema:
         """Generate schema for a NamedTuple."""
-        annotations: dict[str, Any] = get_type_hints_infer_globalns(
-            namedtuple_cls, include_extras=True, localns=self.types_namespace
-        )
-        if not annotations:
-            # annotations is empty, happens if namedtuple_cls defined via collections.namedtuple(...)
-            annotations = {k: Any for k in namedtuple_cls._fields}
+        with self.defs.get_schema_or_ref(namedtuple_cls) as (namedtuple_ref, maybe_schema):
```

Comment:
@dmontagu I think, next time we touch this, we should think about generalizing this with the dataclass schema. The code looks fairly similar.